File: LeetCode/BFS/993. Cousins in Binary Tree/Ingyu.java
import javax.swing.tree.TreeNode;
import java.util.HashSet;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Set;

class Solution {
    public boolean isCousins(TreeNode root, int x, int y) {
        boolean isExist = false;
        int sum = x + y;
        Set<TreeNode> set = new HashSet<>();
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);

        while (!queue.isEmpty()) {
            int size = queue.size();
            while (size-- > 0) {
                TreeNode temp = queue.poll();

                if (temp.left != null) {
                    queue.offer(temp.left);
                    if (temp.left.val == x || temp.left.val == y) {
                        set.add(temp);
                        isExist = true;
                        sum -= temp.left.val;
                    }
                }
                if (temp.right != null) {
                    queue.offer(temp.right);
                    if (temp.right.val == x || temp.right.val == y) {
                        set.add(temp);
                        isExist = true;
                        sum -= temp.right.val;
                    }
                }
            }
            if (isExist == true && sum != 0) return false;
            else if (isExist == true && sum == 0 && set.size() == 2) return true;
        }

        return false;
    }
}

File: LeetCode/BFS/1302. Deepest Leaves Sum/Ingyu.kt
import java.util.*

class Solution {
    fun deepestLeavesSum(root: TreeNode?): Int {
        var sum: Int = 0;
        val q: Queue<TreeNode> = LinkedList<TreeNode>()
        q.offer(root)

        while (!q.isEmpty()) {
            //시작하면 매번 값을 0으로 초기화해서.
            //이전 층의 결과가 이어지지 않게함.
            sum = 0;
            //현재 계층의 요소 구하기.
            var numOfCurrentDepth = q.size
            while (numOfCurrentDepth-- > 0) {
                val temp: TreeNode = q.poll()
                sum += temp.`val`

                temp.left?.let {q.offer(it)}
                temp.right?.let {q.offer(it)}
            }
        }

        return sum
    }
}
